### DISCUSSION TITLE
Typing errors when using `sa_type` and "specific" SQLAlchemy types

### DISCUSSION THREAD

**Discussion by sebastian-correa** [fastapi/sqlmodel#1228] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from datetime import datetime

from sqlalchemy.ext.compiler import compiles
from sqlalchemy.sql import expression
from sqlmodel import (
    DateTime,
    FetchedValue,
    Field,
    SQLModel,
)


### utcnow specific for postgres, to generate the timestamp server side directly in UTC.
# https://stackoverflow.com/a/33532154/8521923
# https://docs.sqlalchemy.org/en/20/core/compiler.html#utc-timestamp-function
class utcnow(expression.FunctionElement):  # noqa: N801
    type = DateTime()
    inherit_cache = True


@compiles(utcnow, "postgresql")
def pg_utcnow(element, compiler, **kw):  # noqa: ARG001 needed for SQLAlchemy compatibility.
    return "TIMEZONE('utc', CURRENT_TIMESTAMP)"


class TimeAuditMixin(SQLModel):
    """Mixin to add created_at and updated_at columns to DB."""

    created_at: datetime | None = Field(
        default=None,
        sa_type=DateTime(timezone=False),
        sa_column_kwargs={"server_default": utcnow(), "nullable": False},
    )
    """Timestamp where the row was inserted to the DB."""

    updated_at: datetime | None = Field(
        default=None,
        sa_type=DateTime(timezone=False),
        sa_column_kwargs={
            "server_default": utcnow(),  # So that inserts from objects without this set don't fail.
            "server_onupdate": FetchedValue(),  # Check https://docs.sqlalchemy.org/en/20/core/defaults.html#triggered-columns
            "nullable": False,
        },
    )
    """Timestamp where the row was last changed."""
```

### Description

I get this type warning:
```
Argument of type "DateTime" cannot be assigned to parameter "sa_type" of type "Type[Any] | PydanticUndefinedType" in function "Field"
  Type "DateTime" is not assignable to type "Type[Any] | PydanticUndefinedType"
    "DateTime" is not assignable to "Type[Any]"
    "DateTime" is not assignable to "PydanticUndefinedType"Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)
```

which I think is because we're not passing `type[Any]` cause we instantiate the `DateTime` to force `timezone=False`. Many people seem to have this issue as well [[ref](https://github.com/fastapi/sqlmodel/discussions/743#discussioncomment-10562815)].

### Operating System

macOS

### Operating System Details

15.1 (24B83) Mac M1

### SQLModel Version

0.0.22

### Python Version

Python 3.10.8

**Comment by iloveitaly** [comment]
Running into this issue as well—not sure the best workaround here.

**Reply by sebastian-correa** [reply]
We've just let the typing thing complain, lest we fill our code with `# type: ignore`s...
